keps/582-preempt-based-on-flavor-order/README.md

```diff
@@ -164,16 +165,52 @@ For each type of resource in each podSet, Kueue will traverse all resource group
 
 ```
 const (
-	Borrow FlavorFungibilityPolicy = "Borrow"
-	Preempt  FlavorFungibilityPolicy = "Preempt"
+  Borrow FlavorFungibilityPolicy = "Borrow"
+  Preempt  FlavorFungibilityPolicy = "Preempt"
   TryNextFlavor FlavorFungibilityPolicy = "TryNextFlavor"
 )
 
+type FlavorSelectionPolicy string
+
+const (
+   PreferBorrowing FlavorSelectionPolicy = "PreferBorrowing"
+   PreferPreemption  FlavorSelectionPolicy = "PreferPreemption"
+)
+
 type FlavorFungibility struct {
-  // +kubebuilder:validation:Enum="Borrow,TryNextFlavor"
+  // whenCanBorrow determines whether a workload should try the next flavor
+  // or stop the search. The possible values are:
+  //
+  // - `Borrow` (default): stop searching and use the best flavor found so far
+  //   according to the selection strategy
+  // - `TryNextFlavor`: try next flavor even if the current
+  //   flavor has enough resources to borrow.
+  //
+  // +kubebuilder:validation:Enum={Borrow,TryNextFlavor}
+  // +kubebuilder:default="Borrow"
   WhenCanBorrow FlavorFungibilityPolicy  `json:"whenCanBorrow"`
-  // +kubebuilder:validation:Enum="Preempt,TryNextFlavor"
+  // whenCanPreempt determines whether a workload should try the next flavor
+  // or stop the search. The possible values are:
+  //
+  // - `Preempt`: stop searching and use the best flavor found so far 
+  //   according to the selection strategy
+  // - `TryNextFlavor` (default): try next flavor even if there are enough
+  //   candidates for preemption in the current flavor.
+  //
+  // +kubebuilder:validation:Enum={Preempt,TryNextFlavor}
+  // +kubebuilder:default="TryNextFlavor"
   WhenCanPreempt FlavorFungibilityPolicy `json:"whenCanPreempt"`
+  // whenCanPreemptAndBorrow defines how should a workload chose the flavor in case of
+  // multiple options. If either borrowing or preemption is necessary and multiple options
+  // are available, then this field defines the selection strategy. The possible values are:
+  //
+  // - `PreferBorrowing` (default): prefer to allocate in a flavor that does not preempt
+  //    other workloads.
+  // - `PreferPreemption`: prefer to allocate in a flavor that uses only the nominal quota
+  //    even if it means preempting other workloads.
+  // +kubebuilder:validation:Enum={PreferBorrowing,PreferPreemption}
```

Comment:
Why not just `Borrow`,`Preempt`? Since the field is already `whenCanPreemptAndBorrow` it is already clear we have the two options.